### DISCUSSION TITLE
Prioritize org members in the "Add people" search bar

### DISCUSSION THREAD

**Discussion by Facuiguacel** [community/community#125588] [Repositories]
### Select Topic Area

Product Feedback

### Body

So whenever I need to give repository access to a specific person from my org, the search bar in "Add people" shows matching names in ALL of github, and if its a common name the person i'm looking for may not even show at all on the list, making me have to type the full first and last name or remember/ask for their username handle. I suggest org members show first even when not the best match for what is being typed. Or, add a toggable filter to search only in org. Specially if it's an org's private repository, might not make sense in public or personal repos.

**Comment by FutureSmart-Pradip** [comment]
Why Cant Github have such basic funcationality

**Comment by sahare-mayur-0071** [comment]
+1 from me.

This becomes particularly frustrating in larger organizations where multiple members share common first names or surnames. In those cases, the current search behavior often surfaces unrelated GitHub users before people who are already members of the organization.

For private repositories, organization members are usually the most likely candidates to receive access, so prioritizing them in search results would significantly reduce administrative friction.

A few possible approaches:

* Rank organization members above external users when searching from an organization-owned repository.
* Add an "Organization members only" filter in the search dialog.
* Display organization members in a separate section at the top of the results.
* Allow searching by team name and then selecting members from that team.

This would improve both usability and security by reducing the risk of accidentally selecting the wrong GitHub account when granting repository access.
